crates/bevy_gizmos/src/primitives/dim3.rs

```diff
@@ -0,0 +1,884 @@
+    position: Vec3, // Center position of the sphere in 3D space
+    color: Color,   // Color of the sphere
+
+    segments: usize, // Number of segments used to approximate the sphere geometry
+}
+
+impl<T: GizmoConfigGroup> SphereBuilder<'_, '_, '_, T> {
+    /// Set the number of segments used to approximate the sphere geometry.
+    pub fn segments(mut self, segments: usize) -> Self {
+        self.segments = segments;
+        self
+    }
+}
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive3d<Sphere> for Gizmos<'w, 's, T> {
+    type Output<'a> = SphereBuilder<'a, 'w, 's, T> where Self: 'a;
+
+    fn primitive_3d(
+        &mut self,
+        primitive: Sphere,
+        position: Vec3,
+        rotation: Quat,
+        color: Color,
+    ) -> Self::Output<'_> {
+        SphereBuilder {
+            gizmos: self,
+            radius: primitive.radius,
+            position,
+            rotation,
+            color,
+            segments: DEFAULT_NUMBER_SEGMENTS,
+        }
+    }
+}
+
+impl<T: GizmoConfigGroup> Drop for SphereBuilder<'_, '_, '_, T> {
+    fn drop(&mut self) {
+        if !self.gizmos.enabled {
+            return;
+        }
+
+        let SphereBuilder {
+            radius,
+            position: center,
+            rotation,
+            color,
+            segments,
+            ..
+        } = self;
+
+        // draw two caps, one for the "upper half" and one for the "lower" half of the sphere
+        [-1.0, 1.0].into_iter().for_each(|sign| {
+            let top = *center + (*rotation * Vec3::Y) * sign * *radius;
+            draw_cap(
+                self.gizmos,
+                *radius,
+                *segments,
+                *rotation,
+                *center,
+                top,
+                *color,
+            );
+        });
+
+        draw_circle(self.gizmos, *radius, *segments, *rotation, *center, *color);
+    }
+}
+
+// plane 3d
+
+/// Builder for configuring the drawing options of [`Sphere`].
+pub struct Plane3dBuilder<'a, 'w, 's, T: GizmoConfigGroup> {
+    gizmos: &'a mut Gizmos<'w, 's, T>,
+
+    normal: Direction3d, // direction of the normal orthogonal to the plane
+
+    rotation: Quat, // Rotation of the sphere around the origin in 3D space
+    position: Vec3, // Center position of the sphere in 3D space
+    color: Color,   // Color of the sphere
+
+    num_axis: usize,     // Number of axis to hint the plane
+    num_segments: usize, // Number of segments used to hint the plane
+    len_segments: f32,   // Length of segments used to hint the plane
+}
+
+impl<T: GizmoConfigGroup> Plane3dBuilder<'_, '_, '_, T> {
+    /// Set the number of segments used to hint the plane.
+    pub fn segments(mut self, segments: usize) -> Self {
+        self.num_segments = segments;
+        self
+    }
+
+    /// Set the length of segments used to hint the plane.
+    pub fn len_segments(mut self, length: f32) -> Self {
+        self.len_segments = length;
+        self
+    }
```

Comment:
This could maybe be fine, but `len` in Rust might give the impression of "the length of a collection" (e.g. `Vec::len`, in this case "the number of segments in this builder") instead of "math vector length".

Glam uses `length` for vector length, so `segment_length` might be clearer
```suggestion
    /// Set the length of the segments used to hint the plane.
    pub fn segment_length(mut self, length: f32) -> Self {
        self.segment_length = length;
        self
    }
```
(this would also match the word order of `segment_count` if we rename `num_segments` to that)